spec/Section 3 -- Type System.md

```diff
@@ -232,6 +232,163 @@ Schema extensions have the potential to be invalid if incorrectly defined.
+    reviewCount: Int
+    friends: [User]
+    email: String @private(scope: "loggedIn")
+    city: String
+}
+
+type Business implements Address {
+    name: String
+    address: String
+    rating: Int
+    city: String
+    reviews: [Review]
+    createdAt: DateTime
+}
+
+type Review {
+    content: String
+    author: User
+    business: Business
+    createdAt: DateTime
+}
+
+union Entity = User | Business | Review
+
+enum SearchFilter {
+    OPEN_NOW
+    DELIVERS_TAKEOUT
+    VEGETARIAN_MENU
+}
+
+type Query {
+    searchBusiness(name: String!, filter: SearchFilter): Business
+}
+
+type Mutation {
+    addReview(input: ReviewInput!): Review
+}
+```
+
+The following table shows examples of Schema Coordinates for elements in the
+schema above:
+
+| Schema Coordinate              | Description                                                         |
+| ------------------------------ | ------------------------------------------------------------------- |
+| `Business`                     | `Business` type                                                     |
+| `User.name`                    | `name` field on the `User` type                                     |
+| `Query.searchBusiness(name:)`  | `name` argument on the `searchBusiness` field on the `Query` type   |
+| `SearchFilter`                 | `SearchFilter` enum                                                 |
+| `SearchFilter.OPEN_NOW`        | `OPEN_NOW` value of the`SearchFilter` enum                          |
+| `@private`                     | `@private` directive definition                                     |
+| `@private(scope:)`             | `scope` argument on the `@private` directive definition             |
+| `Address`                      | `Address` interface                                                 |
+| `Address.city`                 | `city` field on the `Address` interface                             |
+| `ReviewInput`                  | `ReviewInput` input object type                                     |
+| `ReviewInput.author`           | `author` input field on the `ReviewInput` input object type         |
+| `Entity`                       | `Entity` union definition                                           |
+| `DateTime`                     | Custom `DateTime` scalar type                                       |
+| `String`                       | Built-in `String` scalar type                                       |
+
+Schema Coordinates are always unique. Each type, field, argument, enum value, or
+directive may be referenced by exactly one possible Schema Coordinate.
+
+For example, the following is *not* a valid Schema Coordinate:
+
+```graphql counter-example
+Entity.Business
+```
+
+In this counter example, both `Entity.Business` and `Business` would refer to
+the `Business` type. Since it would be ambiguous what the "primary key" should
+be in an application that uses Schema Coordinates to reference types, this is
+not allowed.
```

Comment:
is this waffle? thought it might be helpful to have a little explanation/demonstration of this property

Reply:
Could just be

```diff
- Since it would be ambiguous what the "primary key" should be in an application that uses Schema Coordinates to reference types, this is not allowed.
+ Such ambiguity is disallowed.
```

(More terse, but doesn't walk the reader through _why_ this is bad)

Reply:
> In this counter example, `Entity.Business` is redundant since `Business` already uniquely identifies the `Business` type. Such redundancy is disallowed by this spec - every type, field, field argument, enum value, directive, and directive argument has exactly one canonical Schema Coordinate.

Reply:
👍 thanks benjie, stealing this wording